swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/edda/Edda.kt

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.aws.edda
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.config.EddaApiClient
+import com.netflix.spinnaker.kork.core.RetrySupport
+import com.netflix.spinnaker.security.AuthenticatedRequest
+import com.netflix.spinnaker.swabbie.aws.Parameters
+import com.netflix.spinnaker.swabbie.aws.AWS
+import com.netflix.spinnaker.swabbie.aws.autoscalinggroups.AmazonAutoScalingGroup
+import com.netflix.spinnaker.swabbie.aws.edda.providers.EddaApiSupport
+import com.netflix.spinnaker.swabbie.aws.edda.providers.maxRetries
+import com.netflix.spinnaker.swabbie.aws.edda.providers.retryBackOffMillis
+import com.netflix.spinnaker.swabbie.aws.images.AmazonImage
+import com.netflix.spinnaker.swabbie.aws.instances.AmazonInstance
+import com.netflix.spinnaker.swabbie.aws.launchconfigurations.AmazonLaunchConfiguration
+import com.netflix.spinnaker.swabbie.aws.loadbalancers.AmazonElasticLoadBalancer
+import com.netflix.spinnaker.swabbie.aws.securitygroups.AmazonSecurityGroup
+import com.netflix.spinnaker.swabbie.aws.snapshots.AmazonSnapshot
+import com.netflix.spinnaker.swabbie.model.LAUNCH_CONFIGURATION
+import com.netflix.spinnaker.swabbie.model.SERVER_GROUP
+import com.netflix.spinnaker.swabbie.model.INSTANCE
+import com.netflix.spinnaker.swabbie.model.SNAPSHOT
+import com.netflix.spinnaker.swabbie.model.IMAGE
+import com.netflix.spinnaker.swabbie.model.LOAD_BALANCER
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import retrofit.RetrofitError
+
+class Edda(
+  eddaApiClients: List<EddaApiClient>,
+  private val retrySupport: RetrySupport,
+  private val registry: Registry
+) : AWS, EddaApiSupport(eddaApiClients, registry) {
+  private val log: Logger = LoggerFactory.getLogger(javaClass)
+
+  override fun getLaunchConfigurations(params: Parameters): List<AmazonLaunchConfiguration> {
+    withEddaClient(
+      region = params.region,
```

Comment:
can we make it so `withEddaClient` takes a `Parameters` object instead of duplicating this over and over?

Reply:
Yeah, this class needs to be refactored. Needs to be cleaned up. I will take of this in this PR

Reply:
Going to tackle this in another PR as discussed